src/client/UserPerm.h

```diff
@@ -65,8 +65,8 @@ struct UserPerm
     return *this;
   }
 
-  uid_t uid() const { return m_uid; }
-  gid_t gid() const { return m_gid; }
+  uid_t uid() const { return m_uid != (uid_t)-1 ? m_uid : ::geteuid(); }
+  gid_t gid() const { return m_gid != (gid_t)-1 ? m_gid : ::getegid(); }
```

Comment:
Hmm...maybe I'm wrong there actually. The setfsuid manpage says "don't use it", so maybe geteuid is fine. I think you can disregard my comment...